python-avd/pyavd/schema/metaschema/resolvemodel.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from copy import deepcopy
+from functools import lru_cache
+
+from deepmerge import conservative_merger
+
+from ..constants import STORE
+
+
+def merge_schema_from_ref(schema: dict) -> dict:
+    if "$ref" not in schema:
+        return schema
+
+    schema = deepcopy(schema)
+    ref = schema.pop("$ref")
+    ref_schema = merge_schema_from_ref(get_schema_from_ref(ref))
+    if ref_schema["type"] != schema["type"]:
+        raise ValueError(
```

Comment:
just a note to check later in the process if keeping these generic python errors is ok or if we should add "pyavd centric exceptions"

Reply:
added a todo